src/Paprika/Store/StorageFanOutPage.cs

```diff
@@ -115,6 +115,28 @@ public void Accept(IPageVisitor visitor, IPageResolver resolver, DbAddress addr)
             }
         }
     }
+    public Page Destroy(IBatchContext batch, in NibblePath prefix)
+    {
+        if (Header.BatchId != batch.BatchId)
+        {
+            // the page is from another batch, meaning, it's readonly. Copy
+            var writable = batch.GetWritableCopy(page);
+            return new StorageFanOutPage<TNext>(writable).Destroy(batch, prefix);
+        }
+        var map = new SlottedArray(Data.Data);
+        map.DeleteByPrefix(prefix);
+
+        var index = GetIndex(prefix);
+        ref var addr = ref Data.Addresses[index];
+        Page child;
+        if (!addr.IsNull)
+        {
+            child = batch.GetAt(addr);
+            var sliced = prefix.SliceFrom(ConsumedNibbles);
+            return TNext.Wrap(child).Destroy(batch, sliced);
```

Comment:
No! It's a child. Get its address and store in addr.

Reply:
I didn't get this part.

Reply:
All `Set` and probably `Destroy` should have the same semantics. The page changes something in itself, does COW if needed, and returns the raw `Page` that was modified.

If you `TNext.Wrap(child).Destroy(batch, sliced)` I believe it should be 

```csharp
child = batch.GetAt(addr);
var sliced = prefix.SliceFrom(ConsumedNibbles);
var cowed = TNext.Wrap(child).Destroy(batch, sliced);
addr = batch.GetAt(cowed);
```

assuming that `addr` is a `ref var`

Reply:
Here, as you suggested, I have used COW. Maybe I am returning the wrong output here. Currently, I am returning a new page. Are you suggesting to return the "cowed" page?